lib/req/steps.ex

```diff
@@ -1569,6 +1569,20 @@ defmodule Req.Steps do
     {request, update_in(response.body, &:zlib.gunzip/1)}
   end
 
+  defp decode_body({request, response}, "zst") do
+    if ezstd_loaded?() do
+      case :ezstd.decompress(response.body) do
+        decompressed when is_binary(decompressed) ->
+          {request, put_in(response.body, decompressed)}
+
+        {:error, _} ->
+          {request, %Req.ArchiveError{format: :zstd, data: response.body}}
```

Comment:
zstd is not an archive like tar and zip though, I don't think ArchiveError is appropriate here. please return a RuntimeError with custom message.